scripts/docs_version_pragma_check.sh

```diff
@@ -32,13 +32,114 @@ SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-build}
 source "${REPO_ROOT}/scripts/common.sh"
 source "${REPO_ROOT}/scripts/common_cmdline.sh"
 
+function getAllAvailableVersions()
+{
+    allVersions=()
+    local prevVer
+    local currVer
+    while IFS= read -r line
+    do
+        if (echo "$line" | grep -Po '(?<=soljson-v)\d+.\d+.\d+(?=\+commit)' >/dev/null)
+        then
+            currVer=$(echo "$line" | grep -Po '(?<=soljson-v)\d+.\d+.\d+(?=\+commit)')
+            if [ "$currVer" != "$prevVer" ]
+            then
+                allVersions=( "$currVer" "${allVersions[@]}" )
+                prevVer="$currVer"
+            fi
+            # first available release on github
+            if [ $currVer == "0.4.11" ]
+            then
+                break
+            fi
+        fi
+    done < <(wget -q -O- https://ethereum.github.io/solc-bin/bin/list.txt)
+}
+
+function versionGreater()
+{
+    v1=$1
+    v2=$2
+    ver1=( ${v1//./ } )
+    ver2=( ${v2//./ } )
+
+    if (( ${ver1[0]} > ${ver2[0]} ))
+    then
+        return 0
+    elif (( ${ver1[0]} == ${ver2[0]} )) && (( ${ver1[1]} > ${ver2[1]} ))
+    then
+        return 0
+    elif (( ${ver1[0]} == ${ver2[0]} )) && (( ${ver1[1]} == ${ver2[1]} )) && (( ${ver1[2]} > ${ver2[2]} ))
+    then
+        return 0
+    fi
+    return 1
+}
+
+function versionEqual()
+{
+    if [ "$1" == "$2" ]
+    then
+        return 0
+    fi
+    return 1
+}
+
+function findMinimalVersion()
+{
+    local f=$1
+    local greaterEq=false
+    local greater=false
+    local maxVer=false
+    local pragmaVersion
+
+    # Get minimum compiler version defined by pragma
+    if (grep -Po '(?<=pragma solidity >=)\d+.\d+.\d+' "$f" >/dev/null)
+    then
+        pragmaVersion="$(grep -Po '(?<=pragma solidity >=)\d+.\d+.\d+' "$f")"
+        sign=">="
+        greaterEq=true
+    elif (grep -Po '(?<=pragma solidity \^)\d+.\d+.\d+' "$f" >/dev/null)
+    then
+        pragmaVersion="$(grep -Po '(?<=pragma solidity \^)\d+.\d+.\d+' "$f")"
+        sign="^"
+        maxVer=true
+    elif (grep -Po '(?<=pragma solidity >)\d+.\d+.\d+' "$f" >/dev/null)
+    then
+        pragmaVersion="$(grep -Po '(?<=pragma solidity >)\d+.\d+.\d+' "$f")"
+        sign=">"
+        greater=true;
+    fi
+
+    local minVersion=""
+    for ver in "${allVersions[@]}"
```

Comment:
Can you explain the logic in this loop? I don't understand why `>=` is treated different than `^`.

Reply:
for example 
`pragma solidity >= 0.4.10` will try to compile with 0.4.11
and `pragma solidity ^0.4.10` will skip as there is no 0.4.10 release

Reply:
Other option is to skip in both cases, than it will not defer. Than less checks will be available

Reply:
`^ X` is defined as `>= X <"X+1")`, i.e. `^0.4.10` is equal to `>=0.4.10 <0.5.0` - so wit regards to the oldest version, they should be the same.

Reply:
Ok, than should be the same, updating accordingly